packages/dialtone-css/lib/build/less/components/emoji-text-wrapper.less

```diff
@@ -1,5 +1,17 @@
 .d-emoji-text-wrapper {
-  .d-icon {
-    vertical-align: top;
+  &__text {
+    vertical-align: middle;
+  }
+
+  .d-emoji {
+    position: relative;
+    display: inline-flex;
+    vertical-align: middle;
+
+    .d-icon {
+      position: absolute;
+      top: 50%;
+      transform: translateY(-50%);
+    }
   }
 }
```

Comment:
@juliodialpad, 

OK, here's what I've landed on: https://codepen.io/francisrupert/pen/OJeXjGO?editors=1100

This controls the height of the Emoji itself by inheriting, specifically `height: 1em`. It then positions upward using the age-old `top: 50%; transform: translateY(-50%);`.

So yes, bigger Emoji sizes in smaller text will bleed out... though that will probably only occur if someone **decides** the emoji should be a bigger size and accepts the overlap.

Short screencast walkthrough attached. 

https://github.com/user-attachments/assets/206cc561-e15f-41f9-80f5-a3aa940b13fa

Reply:
Note that I renamed the selector with `.ZZZ-` just to ensure it doesn't collide with any existing `d-emoji-text-wrapper`. Just rename it back and adjust as needed.

Reply:
This is the issue I've been having, and this is why I took several days trying to center this and end up with such a "messy" or "dirty" solution, was the only way I found to actually center the emojis in every case.

<img width="1399" alt="Screenshot 2024-07-23 at 12 28 30 p m" src="https://github.com/user-attachments/assets/4aa3c97d-893e-4438-be3b-1b87058deacc">

I'm seeing the emojis are not actually centered if put lines on the text, you can see the emojis are actually lower than expected.

Reply:
Let's do this...

```
top: calc(50% - 1px);
```

Updated on https://codepen.io/francisrupert/pen/OJeXjGO?editors=1100

![image](https://github.com/user-attachments/assets/f13cb149-8d2b-4897-a55f-ab4c536dfb93)

Reply:
> top: calc(50% - 1px);

Will that actually scale correctly with different text size and emoji sizes??

Reply:
> ... scale correctly with different text sizes and emoji sizes?

Granted, an absolute nudging value of `1px` may feel odd across all the sizes, it's quite consistent across varying emoji sizes within varying text size and line-height. Can stress-test it at https://codepen.io/francisrupert/pen/OJeXjGO?editors=1100

What I'm most concerned about is shifting baselines and line-height of the parent element or elements.

**Optical** alignment is a finicky thing, so it becomes a design judgement call. While a few of these may not _strictly_ be perfectly centered, their optical alignment is pretty darn good. Additionally, the few that barely aren't perfectly centered aren't likely to occur in practical use — much like bigger emojis won't be used within small text size. 

![image](https://github.com/user-attachments/assets/bca0d075-8faf-49ad-b9c2-f8ad07b70d12)

Reply:
Alright, I'll do the changes and see how it behaves on visual tests.